### PR TITLE
Update the tree-sitter dependency

### PR DESCRIPTION
**Pull request by icecream17**
### Description of the Change

Updates tree-sitter-java to v19

Or rather, a fork of tree-sitter-java.
So now I made a fork that combines v19 and the old fork together.

https://github.com/tree-sitter/tree-sitter-java/compare/master...icecream17:cd65bb4cce7b2ba4f444aadd2202b12c484de036
https://github.com/tree-sitter/tree-sitter-java/compare/master...sadikovi:master

My fork passes the [tests on the tree-sitter-java repo](https://github.com/icecream17/tree-sitter-java/runs/5026487832),
but _not [the tests on this repo](https://github.com/icecream17/language-java/runs/5026539450)_.

Part of https://github.com/atom/atom/issues/22129 and https://github.com/icecream17/atom-update-backlog/blob/main/Languages.md

Don't merge yet

### Applicable Issues

https://github.com/atom/atom/issues/22129

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "url": "https://github.com/atom/language-java/issues"
   },
   "dependencies": {
-    "tree-sitter-java-dev": "^0.16.0-dev2"
+    "tree-sitter-java-dev": "git+https://github.com/icecream17/tree-sitter-java.git#cd65bb4cce7b2ba4f444aadd2202b12c484de036"
```
Comment (sadikovi) [inline package.json:19]: Is this for testing only for now? Are there any plans to move the changes to the fork?

**Comment by sadikovi** [comment]
The tree-sitter fork actually has some bug fixes that are not in the main repo. Maybe it would be easier to rebase the fork 🤔.

**Comment by icecream17** [comment]
Yeah rebasing would be better.

I tried to keep the bug fixes in the fork but it seems like somehow the tests are broken (it's either the updates or I did something wrong)

Also, just rebasing doesn't work, edit: because of the regenerate command?

https://github.com/tree-sitter/tree-sitter-java/compare/master...icecream17:cd65bb4cce7b2ba4f444aadd2202b12c484de036
https://github.com/tree-sitter/tree-sitter-java/compare/master...sadikovi:master

Unkept changes: Identifier (it was fixed more thoroughly in the main repo)
New changes: Updating queries

**Comment by sadikovi** [comment]
Thanks, rebasing should have worked 🤔  although it might require conflict resolution. Either way, would you be able to fix the tests?

**Comment by icecream17** [comment]
I'm not sure why the tests are failing right now 😅

**Comment by sadikovi** [comment]
You can run the tests locally with `apm test` (it takes approximately 10-20sec) and check if they are failing there.

**Comment by darangi** [comment]
@icecream17 are you still looking into this?

**Comment by icecream17** [comment]
Not right now; feel free to close this

**Pull request closed without merging**